R/plotSpectra-PTM.R

```diff
@@ -0,0 +1,482 @@
+
+    if (deltaMz) {
+        deltaMzData <- labelFragments(x, ppm = ppm, what = "mz", ...)
+        layout_matrix <- .make_layout_matrix(length(labels))
+        layout(layout_matrix,
+               heights = rep(c(5, 1), length.out = nrow(layout_matrix)))
+    } else {
+      par(mfrow = n2mfrow(length(labels), asp = asp))
+      deltaMzData <- NULL
+      }
+    spectrum_number <- attr(labels, "group")
+    
+    for (i in seq_along(spectrum_number)) {
+        .plot_single_spectrum_PTM(x[spectrum_number[i]], xlab = xlab, 
+                                  ylab = ylab, xlim = xlim, 
+                                  ylim = ylim, 
+                                  main = main[spectrum_number[i]],
+                                  col = col, labels = labels[i],
+                                  labelCex = labelCex, labelSrt = labelSrt,
+                                  labelAdj = labelAdj, labelPos = labelPos,
+                                  labelOffset = labelOffset, 
+                                  minorTicks = minorTicks, 
+                                  deltaMzData = deltaMzData[[i]],
+                                  ppm = ppm)
+    }
+    on.exit(par(old_par))
+}
+
+
+#' @description
+#'
+#' Plot a single spectrum (m/z on x against intensity on y) whilst 
+#' labeling the individual peaks based on `labelFragments()`.
+#'
+#' @author Johannes Rainer, Sebastian Gibb, Guillaume Deflandre, Laurent Gatto
+#'
+#' @importFrom graphics axis plot.new plot.window plot.xy strwidth
+#'
+#' @importFrom graphics text title
+#'
+#' @importFrom grDevices xy.coords
+#' 
+#' @noRd
+.plot_single_spectrum_PTM <- function(x, xlab = "m/z", ylab = "intensity",
+                                      xlim = numeric(),
+                                      ylim = numeric(), main = character(),
+                                      col = c(y = "darkred",
+                                              b = "darkblue",
+                                              acxy = "darkgreen", 
+                                              other = "grey40"),
+                                      labels = list(),
+                                      labelCol = col, labelCex = 1, labelSrt = 0,
+                                      labelAdj = NULL, labelPos = 3,
+                                      labelOffset = 0.5, minorTicks = TRUE,
+                                      ppm = 20, deltaMzData = NULL) {
+    v <- peaksData(x)[[1L]]
+    mzs <- v[, "mz"]
+    ints <- v[, "intensity"]
+    old_par <- par(no.readonly = TRUE)
+    
+    if (!length(xlim))
+        suppressWarnings(xlim <- range(mzs, na.rm = TRUE))
+    if (!length(ylim))
+        suppressWarnings(
+            ylim <- range(c(0, max(abs(ints), na.rm = TRUE))))
+    if (any(is.infinite(xlim)))
+        xlim <- c(0, 0)
+    if (any(is.infinite(ylim)))
+        ylim <- c(0, 0)
+    if (length(main)) {
+        par(mar = c(4,4,1.4,2))
+    } else {par(mar = c(4,4,0.5,2))}
```

Comment:
Fix the syntax here: either `par` is on the same line as `else` and no `{}`, or on the next line with `{}`.